src/staking/index.ts

```diff
@@ -103,20 +103,22 @@ export class StakingPoolService {
    * @param org ENS name of organization
    */
   async getPool(org: string): Promise<StakingPool | null> {
-    const service = await this._stakingPoolFactory.services(namehash(org));
-    if (service.pool === emptyAddress) {
+    const { pool } = await this._stakingPoolFactory.services(namehash(org));
+    if (pool === emptyAddress) {
       return null;
     }
-    const pool = new StakingPool__factory(this._signer).attach(service.pool);
-    return new StakingPool(pool);
+    return new StakingPool(this._signer, pool);
   }
 }
 
 /**
  * Abstraction over staking pool smart contract
  */
 export class StakingPool {
-  constructor(private pool: StakingPoolContract) { }
+  private pool: StakingPoolContract;
+  constructor(private patron: Required<Signer>, address: string) {
```

Comment:
Just curious, why the "Required<>" around the Signer? Does Signer have optional parameters that you want to ensure are set?

Reply:
Correct. In order patron to send transaction he must be connected, but `provider` property of Signer is optional